daemon/config/config.go

```diff
@@ -398,7 +398,7 @@ func getConflictFreeConfiguration(configFile string, flags *pflag.FlagSet) (*Con
 	if flags != nil {
 		var jsonConfig map[string]interface{}
 		reader = bytes.NewReader(b)
-		if err := json.NewDecoder(reader).Decode(&jsonConfig); err != nil {
+		if err := json.NewDecoder(reader).Decode(&jsonConfig); err != nil && err.Error() != "EOF" {
```

Comment:
Better to use `io.EOF` here instead of string matching

```suggestion
		if err := json.NewDecoder(reader).Decode(&jsonConfig); err != nil && err != io.EOF {
```